src/EmbedIO/Utilities/IPParser.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace EmbedIO.Utilities
+{
+    /// <summary>
+    /// Provides standard methods to parse IP address strings.
+    /// </summary>
+    public static class IPParser
+    {
+        /// <summary>
+        /// Parses the specified IP address.
+        /// </summary>
+        /// <param name="address">The IP address.</param>
+        /// <returns>A collection of <see cref="IPAddress"/> parsed correctly from <paramref name="address"/>.</returns>
+        public static async Task<IEnumerable<IPAddress>> Parse(string address)
+        {
+            if (address == null)
+                return Enumerable.Empty<IPAddress>();
+
+            if (IPAddress.TryParse(address, out var ip))
+                return new List<IPAddress> { ip };
+
+            try
+            {
+                return await Dns.GetHostAddressesAsync(address).ConfigureAwait(false);
+            }
+            catch
+            {
+                // Ignore
+            }
+
+            if (IsCIDRNotation(address))
+                return ParseCIDRNotation(address);
+
+            return IsSimpleIPRange(address) ? TryParseSimpleIPRange(address) : Enumerable.Empty<IPAddress>();
+        }
+
+        /// <summary>
+        /// Determines whether the IP-range string is in CIDR notation.
+        /// </summary>
+        /// <param name="range">The IP-range string.</param>
+        /// <returns>
+        ///   <c>true</c> if the IP-range string is CIDR notation; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsCIDRNotation(string range)
```

Comment:
As per good practices, CIDR being a four-letter acronym, it's better not written in all capitals inside identifiers. Therefore, `IsCidrNotation` would be a better name.

Reply:
I already changed the method's name.